active-rfcs/0000-vtu-deprecations.md

```diff
@@ -0,0 +1,188 @@
+# API Deprecation [RFC]
+
+Created: Mar 28, 2020 7:39 PM
+
+- Start Date: 2020.03.28
+- Target Major Version: 1.x
+- Reference Issues:
+- Implementation PR:
+
+# Summary
+
+Deprecate lesser used or bad practice methods and properties from the upcoming VueTestUtils 1.x API, aimed at Vue 3. 
+
+## Motivation
+
+With the current version of Vue Test Utils-beta.x, we notice the API is large and somewhat confusing to most  users, which are not well acquainted with testing Vue apps, leading to bad testing practices. 
+
+This PR aims to trim down the VTU API, replacing listed methods with improved docs, more examples for all the related cases or improved testing methods.
+
+## Detailed design
+
+### Methods
+
+**Wrapper.emittedByOrder -** [Link](https://vue-test-utils.vuejs.org/api/wrapper/#emittedbyorder)
+
+Rarely used, use `emitted` instead.
+
+```js
+expect(wrapper.emitted('change')[0]).toEqual(['param1', 'param2'])
+```
+
+**get -** [Link](https://vue-test-utils.vuejs.org/api/wrapper/#get) 
+
+Recently added, throws error if nothing is matched. Will be merged into `find`.
+
+```js
+expect(wrapper.find('.notExisting')).toThrow()
+
+const element = wrapper.find('.notExisting') // will throw error
```

Comment:
I think that from a testing perspective this is lacking expressiveness: it's not clear what we are testing when we write `expect(wrapper.find('.notExisting')).toThrow()`, specially compared to `expect(wrapper.find('.notExisting').exists()).toBe(false)`

Why do you think throwing an error directly on `find` makes sense? I find the existing behavior of throwing when using any method that assumes the returned wrapper not to be empty like 

```js
let button = wrapper.find('.not-existing-button')
button.classes() // throws
```

more reasonable. With the given change, the test above does not need any change as it will just throw earlier, but I don't find reasonable having to change

```js
expect(wrapper.find('.notExisting').exists()).toBe(false)
```

to

```js
expect(() => wrapper.find('.notExisting')).toThrow()
```

BTW, we do need to create a function wrapper to use `toThrow`, don't we?

Reply:
You are right. This was aimed at merging `get` and `find`. 

`get` was added recently, as people decided throwing an error was better and more intuitive, than checking for `exists` all the time.

The VTU team is a bit uncertain about this.

Reply:
I can live with keeping `get` (although I think just keeping `find` the same as using `expect(wrapper.find('.notExisting').exists()).toBe(false)` is fine).